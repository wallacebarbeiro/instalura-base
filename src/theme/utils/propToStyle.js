import breakpointsMedia from './breakpointsMedia';

function propToStyle(propName) {
  // eslint-disable-next-line func-names
  return function (props) {
    const propValue = props[propName];

    if (typeof propValue === 'string' || typeof propValue === 'number') {
      return {
        // textAlign : props.textAlign
        [propName]: props[propName],
      };
    }

    if (typeof propValue === 'object') {
      return breakpointsMedia({
        xs: {
          [propName]: propValue.xs,
        },
        sm: {
          [propName]: propValue.sm,
        },
        md: {
          [propName]: propValue.md,
        },
        lg: {
          [propName]: propValue.lg,
        },
        xl: {
          [propName]: propValue.xl,
        },
      });
    }
    return this;
  };
}

export default propToStyle;
