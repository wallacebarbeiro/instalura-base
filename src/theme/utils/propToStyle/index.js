import breakpointsMedia from '../breakpointsMedia';

function propToStyle(propName) {
  // eslint-disable-next-line func-names
  return function (props) {
    const propValue = props[propName];

    if (typeof propValue === 'string' || typeof propValue === 'number') {
      return {
        // textAlign : props.textAlign
        [propName]: props[propName],
      };
    }

    if (typeof propValue === 'object') {
      const breakpoints = {};

      if (propValue.xs) breakpoints.xs = { [propName]: propValue.xs };
      if (propValue.md) breakpoints.md = { [propName]: propValue.md };
      if (propValue.lg) breakpoints.lg = { [propName]: propValue.lg };

      return breakpointsMedia(breakpoints);

      //     return breakpointsMedia({
      //       xs: {
      //         [propName]: propValue.xs,
      //       },
      //       sm: {
      //         [propName]: propValue.sm,
      //       },
      //       md: {
      //         [propName]: propValue.md,
      //       },
      //       lg: {
      //         [propName]: propValue.lg,
      //       },
      //       xl: {
      //         [propName]: propValue.xl,
      //       },
      //     });
    }
    return this;
  };
}

export default propToStyle;
