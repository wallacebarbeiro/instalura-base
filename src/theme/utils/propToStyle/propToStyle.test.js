import propToStyle from './index';

describe('propToStyle()', () => {
  describe('when receives a simple argument', () => {
    test(' and it is a string', () => {
    // Vamos inserir o que teremos que testar dentro do compoment antes de
    // passar o objeto com o mesmo valor
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlig="center" />
      const componentProps = { textAlign: 'center' }; // String
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ textAlign: 'center' });
    });
    test(' and it is a number', () => {
    // Vamos inserir o que teremos que testar dentro do compoment antes de
    // passar o objeto com o mesmo valor
      const propToStyleResult = propToStyle('flex');

      // <Text flex=1 />
      const componentProps = { flex: 1 }; // Number
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when receives an orgument with  breakpoints', () => {
    test('renders only one breakpoint resolution', () => {
    // Vamos inserir o que teremos que testar dentro do compoment antes de
    // passar o objeto com o mesmo valor
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlig="center" />
      const componentProps = { textAlign: { xs: 'center' } }; // String
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();

      expect(true).toBe(true);
    });

    test('renders two or more breakpoint resolution', () => {
      // Vamos inserir o que teremos que testar dentro do compoment antes de
      // passar o objeto com o mesmo valor
      const propToStyleResult = propToStyle('textAlign');

      // <Text textAlig="center" />
      const componentProps = { textAlign: { xs: 'center', md: 'right' } }; // String
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();

      expect(true).toBe(true);
    });
  });
});
