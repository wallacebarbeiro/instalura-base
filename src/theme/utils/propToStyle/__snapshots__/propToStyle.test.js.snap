// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when receives an orgument with  breakpoints renders only one breakpoint resolution 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`propToStyle() when receives an orgument with  breakpoints renders two or more breakpoint resolution 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "text-align: right;",
    "}",
  ],
]
`;
